proxyd/backend_test.go

```diff
@@ -20,3 +26,34 @@ func TestStripXFF(t *testing.T) {
 		assert.Equal(t, test.out, actual)
 	}
 }
+
+func TestForwardContextCanceled(t *testing.T) {
```

Comment:
Looks good, would it be possible to add a few more tests and cases such as incrementing the intermittentErrorsSlidingWindow first and then checking that they are not incremented again. 

Maybe asset that the debug log exists too in the case of context canceled.

Also test the multicall routing strategy